README.md

```diff
@@ -255,12 +255,53 @@ convention = "google"
 case-sensitive = true
 lines-after-imports = 2
 
+[tool.ruff.lint.flake8-annotations]
+suppress-dummy-args = true
+
 [tool.ruff.lint.extend-per-file-ignores]
-"tests/*" = ["D100", "D103", "ANN"]
+"*" = [
+    "D1", # Ignore warnings about missing docstrings
+    "ANN401", # Ignore warnings about use of the Any type
+]
+"tests/*" = ["ANN"]
 ```
 
+Notes about the [pydocstyle (D)](https://docs.astral.sh/ruff/rules/#pydocstyle-d) extension:
+
+- Our recommended configuration ignores the `D1` family of errors for missing docstrings in public functions, classes, etc.
+  You may want to remove this from the ignore list, especially if your project is published as a library.
+
+Notes about the [flake8-annotations (ANN)](https://docs.astral.sh/ruff/rules/#flake8-annotations-ann) extension:
+
+- Checks for missing type annotations, but does not perform static type checking (that's mypy's job).
+
+- Mypy also includes options for enforcing type annotations
+  (see [`disallow_untyped_defs`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_untyped_defs)
+  and [`disallow_incomplete_defs`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_incomplete_defs)),
+  but we started using ruff's `ANN` extension before mypy, so we're sticking with `ANN` for now.
+
+- Feel free to remove this extension if you do not want to use static typing for your project.
+  Alternatively, you may want to enable the
+  [`ignore-fully-untyped`](https://docs.astral.sh/ruff/settings/#lint_flake8-annotations_ignore-fully-untyped) option
+  (similar to mypy's [`disallow_incomplete_defs`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_incomplete_defs) option)
+  when adding the `ANN` extension to an existing codebase with portions of untyped code.
+
 ### [`reusable-mypy.yml`](./.github/workflows/reusable-mypy.yml)
 
+> [!WARNING]
+> Mypy does not turn Python into a statically-typed language,
+> especially when mixing typed and untyped code, using the `Any` type,
+> or importing untyped third-party libraries.
+> Type annotations are more like hints about what types values *should* have,
+> rather than a guarantee of runtime type safety.
+> They are not a replacement for best practices when using a dynamically-typed language.
+> For more information about how mypy accommodates dynamic typing,
+> see the docs on [dynamic typing](https://mypy.readthedocs.io/en/stable/dynamic_typing.html)
+> and the [`Any` type](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#the-any-type).
+>
+> Additionally, please review our [Mypy wiki article](https://github.com/ASFHyP3/.github-private/wiki/Mypy)
+> for configuration tips and troubleshooting beyond what's shown here.
```

Comment:
ASFHyP3/Actions is used externally, so we shouldn't be pointing to a private wiki in this README.